lint_test/member_ordering_test.dart

```diff
@@ -0,0 +1,64 @@
+// ignore_for_file: prefer_const_declarations
+// ignore_for_file: unnecessary_nullable_for_final_variable_declarations
+// ignore_for_file: unnecessary_cast
+// ignore_for_file: unused_local_variable
+
+import 'package:flutter/widgets.dart';
+
+/// Check the `member-ordering` rule
+
+class A {
+  int _a = 1;
+  // expect_lint: member-ordering
+  final b = 2;
+
+  // Allowed because it's before constructor according to config
+  set a(int value) => _a = value;
+
+  // expect_lint: member-ordering
+  A();
+
+  void close() {}
+
+  // should be before close-method
+  // expect_lint: member-ordering
+  int get a => _a;
+}
+
+class AWidget extends StatefulWidget {
+  const AWidget({super.key});
+
+  @override
+  State<AWidget> createState() => _AWidgetState();
+}
+
+class _AWidgetState extends State<AWidget> {
```

Comment:
Needs more fleshed-out testcases. Private fields, private methods, static members.
It would be much better to have _all_ that we enforce here.

Reply:
Added more. But technically we can't test all because it has named-method pattern so it's possible to have config like
```
-a-method
-b-method
-c-method
-d-method
...
```

Reply:
I agree that testing all is not viable, but I would still ask for more.

Let's structure the tests a bit. Here's what I think will be helpful:

1. One all-positive case where we should see no lints at all.
2. One negative case with all ticks triggered.
3. One semi-negative case that would check that we can detect one isolated lint without triggering anything unrelated.

I'd also create two separate sets of tests for regular classes and for widgets, so there'll be 6 classes.

Reply:
Done

Reply:
I also see some code that would enfoce alphabetic order and alphabetic-by-type order. I don't see any tests for that -- just a couple basic ones will suffice.

Reply:
I can't test them both because you can pass only one of them at `analysis_options.yaml` will add for alphabetical.

Reply:
If we can't test it -- then it doesn't work. Let's try to find a way.
One option would be to create a subdir with its own `analysis_options.yaml` file -- but I'm not sure custom_lint handles such nesting well. Give it a test.
We can also create a completely separate package with its own config, with no nesting. A bit ugly, but at least we'll have it tested.
I don't think this will be the last rule with potentially conflicting configs.